### PR TITLE
DROID-2121 Relations | Create or edit options screen

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt [hunk 1]
```diff
@@ -25,7 +33,133 @@ class CreateOrEditOptionViewModel(
...
+                            option = option
+                        )
+                    },
+                    failure = { Timber.e(it, "Error while creating option") }
+                )
+            }
+        }
+    }
+
+    private fun updateOption() {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:104]: proceedWithUpdatingOption

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt [hunk 2]
```diff
@@ -25,7 +33,133 @@ class CreateOrEditOptionViewModel(
...
+    }
+
+    fun onButtonClick() {
+        when (viewState.value) {
+            is CreateOrEditOptionScreenViewState.Create -> createOption()
+            is CreateOrEditOptionScreenViewState.Edit -> updateOption()
+        }
+    }
+
+    private fun createOption() {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:80]: proceedWithCreatingOption

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt [hunk 3]
```diff
@@ -25,7 +33,133 @@ class CreateOrEditOptionViewModel(
...
+        viewModelScope.launch {
+            val params = CreateRelationOption.Params(
+                space = spaceManager.get(),
+                relation = viewModelParams.relationKey,
+                name = viewState.value.text,
+                color = viewState.value.color.code
+            )
+            if (params.name.isNotEmpty()) {
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:88]: maybe show toast with "Please enter name for new option" in the `else` case?

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt [hunk 4]
```diff
@@ -25,7 +33,133 @@ class CreateOrEditOptionViewModel(
...
+        }
+    }
+
+    private fun updateOption() {
+        val optionId = viewModelParams.optionId ?: return
+        viewModelScope.launch {
+            val params = SetObjectDetails.Params(
+                ctx = optionId,
+                details = mapOf(
+                    Relations.NAME to viewState.value.text,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:110]: what if text is empty? relevant: https://github.com/anyproto/anytype-kotlin/pull/821/files#r1478393573

**Review by uburoiubu — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt (hunk 3 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:88]: In a case of an empty title user can see a hint "Enter name..."

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt (hunk 2 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:80]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt (hunk 1 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/relations/option/CreateOrEditOptionViewModel.kt:104]: done

**Pull request merged**